standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java

```diff
@@ -5685,6 +5706,67 @@ private void alter_table_core(String catName, String dbname, String name, Table
       }
     }
 
+    @Override
+    public GetFileListResponse get_file_list(GetFileListRequest req) throws MetaException {
+      String catName = req.isSetCatName() ? req.getCatName() : getDefaultCatalog(conf);
+      String dbName = req.getDbName();
+      String tblName = req.getTableName();
+      List<String> partitions = req.getPartVals();
+      // Will be used later, when cache is introduced
+      String validWriteIdList = req.getValidWriteIdList();
+
+      startFunction("get_file_list", ": " + TableName.getQualified(catName, dbName, tblName)
+              + ", partitions: " + partitions.toString());
+
+
+      GetFileListResponse response = new GetFileListResponse();
+
+      boolean success = false;
+      Exception ex = null;
+      try {
+        Partition p =  getMS().getPartition(catName, dbName, tblName, partitions);
+        Path path = new Path(p.getSd().getLocation());
+
+        FileSystem fs = path.getFileSystem(conf);
+        RemoteIterator<LocatedFileStatus> itr = fs.listFiles(path, true);
+        while (itr.hasNext()) {
+          FileStatus fStatus = itr.next();
+          Reader reader = OrcFile.createReader(fStatus.getPath(), OrcFile.readerOptions(fs.getConf()));
+          boolean isRawFormat  = !CollectionUtils.isEqualCollection(reader.getSchema().getFieldNames(), ALL_ACID_ROW_NAMES);
+          int fileFormat = isRawFormat ? 0 : 2;
+          response.addToFileListData(createFileStatus(fStatus, p.getSd().getLocation(), fileFormat));
+        }
+        success = true;
+      } catch (Exception e) {
+        ex = e;
+        LOG.error("Failed to list files with error: " + e.getMessage());
+        throw new MetaException(e.getMessage());
+      } finally {
+        endFunction("get_file_list", success, ex);
+      }
+      return response;
+    }
+
+    private ByteBuffer createFileStatus(FileStatus fileStatus, String basePath, int fileFormat) {
```

Comment:
Please add a java doc here. Also, would be good to attribute the source where this code is from with a link to the file.

Reply:
Noted, I'll add javadoc here.